### PR TITLE
feat: wishlist

### PR DESCRIPTION
**Pull request by Anish-JS**
Added wishlist. Please review.
You can review it after logging or signing up.
For login use --> email: test@gmail.com, password: test123
For signup you can provide your own data

You can add, remove and view the wishList

### CODE REVIEW

**Comment by samflab** [comment]
I would like to point out a few things: 
1. after I click on the wishlist icon, the button doesn't toggle so I don't know if it gets added to the wishlist.
2. if I click multiple times on an item, duplicate products get added. 
3. on hover of the `add to cart` button, it goes blank so fix that.
4. on that navbar, the count of the wishlist item should also be displayed as a badge on the heart icon.

File: src/components/WishList/WishList.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const WishList = () => {
+  const { isLoggedIn } = useAuth();
+  const { showWishListHandler, wishList } = useWishList();
+  useEffect(() => {
+    showWishListHandler();
+
+    return () => {};
+  }, []);
+
+  const wishListMap = wishList.map((item) => <WishCard data={item}></WishCard>);
```
Comment (shobit1337) [inline src/components/WishList/WishList.js]: You need to provide key prop here

File: src/components/WishList/WishList.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  return (
+    <>
+      <div class="section-wishlist sm-padding bg-dull">
```
Comment (shobit1337) [inline src/components/WishList/WishList.js]: Convert these class attributes to className

File: src/store/wishlist-context.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            authorization: localStorage.getItem("token"),
+          },
+        })
+        .then((res) => setWishList(res.data.wishlist));
+    } catch (error) {
+      console.log(error);
+    }
+  };
+  const addToWishListHandler = async (product) => {
+    // console.log(wishList);
```
Comment (shobit1337) [inline src/store/wishlist-context.js]: Remove these unnecessary commented codes.

File: src/components/Card/WishCard.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            className="card__heading para-md semibold-weight center"
+            to={`/product/${props.data["_id"]}`}
+          >
+            {props.data.name}
+          </Link>
+          <div className="card__author para-sm bold">
+            <img src={rupeeIcon} alt="rupee" className="img-xxxs" />
+            <div className="xxs-padding">{props.data.price}</div>
+          </div>
+          {/* <div className="card__info para-sm-xs">{props.data.productInfo}</div> */}
```
Comment (shobit1337) [inline src/components/Card/WishCard.js]: Remove commented code.

**Review by shobit1337 — CHANGES_REQUESTED** [review]

**Comment by shobit1337** [comment]
I have suggested some changes. Also, there are multiple error messages showing in the console.

There is one bug I noticed, We are able to add multiple copies of product in the wishlist from wishlist page itself.

File: src/components/Card/WishCard.js (hunk 4 above)
Comment (Anish-JS) [reply src/components/Card/WishCard.js]: All the changes have been made. Please review it

**Comment by Anish-JS** [comment]
> I have suggested some changes. Also, there are multiple error messages showing in the console.
> 
> There is one bug I noticed, We are able to add multiple copies of product in the wishlist from wishlist page itself.

All changes are made. Please review it

**Comment by Anish-JS** [comment]
> I would like to point out a few things:
> 
> 1. after I click on the wishlist icon, the button doesn't toggle so I don't know if it gets added to the wishlist.
> 2. if I click multiple times on an item, duplicate products get added.
> 3. on hover of the `add to cart` button, it goes blank so fix that.
> 4. on that navbar, the count of the wishlist item should also be displayed as a badge on the heart icon.

All changes are made can you please review it

**Review by samflab — APPROVED** [review]
looks better now.

File: src/components/Card/WishCard.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faHeart, faTimesCircle } from "@fortawesome/free-solid-svg-icons";
+import rupeeIcon from "../../assets/rupee.svg";
+import "./Card.css";
+import { Link } from "react-router-dom";
+
+import { useWishList } from "../../store/wishlist-context";
+import { useState } from "react";
+// import { useCart } from "../../store/cart-context";
```
Comment (Vikas2426) [inline src/components/Card/WishCard.js:11]: remove commented code

**Review by Vikas2426 — APPROVED** [review]

**Comment by shobit1337** [comment]
> Looks good to merge!

**Review by shobit1337 — APPROVED** [review]

**Pull request merged**